src/overlay/Peer.cpp

```diff
@@ -1105,10 +1191,52 @@ Peer::recvGetTxSet(StellarMessage const& msg)
         }
 
         auto newMsgPtr = std::make_shared<StellarMessage const>(newMsg);
+        CLOG_INFO(
+            Overlay,
+            "Peer::recvGetTxSet {} found the tx set for {}, sending it back",
+            toString(), hexAbbrev(msg.txSetHash()));
         self->sendMessage(newMsgPtr);
     }
     else
     {
+        auto slotIndex =
+            mApp.getHerder().getLastSeenSlotIndexForTxSet(msg.txSetHash());
+
+        if (slotIndex == 0)
+        {
+            CLOG_INFO(Overlay,
```

Comment:
I think we should just send DONT_HAVE immediately here, since we're not tracking this tx set.

Reply:
I think the issue is what if malicious nodes send too many malformed requests? If we reply to every message we might flood the network?